src/Symfony/Component/Security/Core/Authorization/SwitchUserAuthorizationChecker.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization;
+
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Role\SwitchUserRole;
+
+/**
+ * SwitchUserAuthorizationChecker.
+ *
+ * It gives access to the token representing the impersonated user.
+ *
+ * @author Fabien Papet <fabien.papet@gmail.com>
+ */
+class SwitchUserAuthorizationChecker implements SwitchUserAuthorizationCheckerInterface
+{
+    /**
+     * @var AccessDecisionManagerInterface
+     */
+    private $accessDecisionManager;
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @param AccessDecisionManagerInterface $accessDecisionManager
+     * @param TokenStorageInterface          $tokenStorage
+     */
+    public function __construct(AccessDecisionManagerInterface $accessDecisionManager, TokenStorageInterface $tokenStorage)
+    {
+        $this->accessDecisionManager = $accessDecisionManager;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function isGranted($attributes, $subject = null)
+    {
+        $token = $this->getOriginalToken($this->tokenStorage->getToken());
+
+        if (false !== $token) {
+            return $this->accessDecisionManager->decide($token, $attributes, $subject);
```

Comment:
maybe checking for `=== false` first and `return false`? Not sure if its worth to switch

Reply:
IMO actual is better than switching which would give something like this

```
        $token = $this->getOriginalToken($this->tokenStorage->getToken());

        if (false === $token) {
            return false;
        }

        return $this->accessDecisionManager->decide($token, $attributes, $subject);

```